api/src/main/java/com/okta/idx/sdk/api/wrapper/AuthenticationWrapper.java

```diff
@@ -0,0 +1,269 @@
+ */
+package com.okta.idx.sdk.api.wrapper;
+
+import com.okta.commons.lang.Assert;
+import com.okta.idx.sdk.api.client.IDXClient;
+import com.okta.idx.sdk.api.exception.ProcessingException;
+import com.okta.idx.sdk.api.model.AuthenticationOptions;
+import com.okta.idx.sdk.api.model.AuthenticationStatus;
+import com.okta.idx.sdk.api.model.ChangePasswordOptions;
+import com.okta.idx.sdk.api.model.Credentials;
+import com.okta.idx.sdk.api.model.FormValue;
+import com.okta.idx.sdk.api.model.IDXClientContext;
+import com.okta.idx.sdk.api.model.RemediationOption;
+import com.okta.idx.sdk.api.model.RemediationType;
+import com.okta.idx.sdk.api.request.AnswerChallengeRequest;
+import com.okta.idx.sdk.api.request.AnswerChallengeRequestBuilder;
+import com.okta.idx.sdk.api.request.IdentifyRequest;
+import com.okta.idx.sdk.api.request.IdentifyRequestBuilder;
+import com.okta.idx.sdk.api.response.AuthenticationResponse;
+import com.okta.idx.sdk.api.response.IDXResponse;
+import com.okta.idx.sdk.api.response.TokenResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class AuthenticationWrapper {
+
+    private static final Logger logger = LoggerFactory.getLogger(AuthenticationWrapper.class);
+
+    /**
+     * Authenticate user with the supplied Authentication options (username and password) and
+     * returns the Authentication response object that contains:
+     * - IDX Client context
+     * - Token (access_token/id_token/refresh_token) object
+     * - Authentication status
+     * <p>
+     * Note: This requires 'Password' as the ONLY required factor in app Sign-on policy configuration.
+     *
+     * @param client                the IDX Client
+     * @param authenticationOptions the Authenticator options
+     * @return the Authentication response
+     */
+    public static AuthenticationResponse authenticate(IDXClient client, AuthenticationOptions authenticationOptions) {
+
+        AuthenticationResponse authenticationResponse = new AuthenticationResponse();
+        TokenResponse tokenResponse;
+        IDXClientContext idxClientContext;
+
+        try {
+            idxClientContext = client.interact();
+            Assert.notNull(idxClientContext, "IDX client context may not be null");
+            authenticationResponse.setIdxClientContext(idxClientContext);
+
+            IDXResponse introspectResponse = client.introspect(idxClientContext);
+            String stateHandle = introspectResponse.getStateHandle();
+            Assert.hasText(stateHandle, "State handle may not be null");
+
+            RemediationOption[] remediationOptions = introspectResponse.remediation().remediationOptions();
+            printRemediationOptions(remediationOptions);
+
+            Optional<RemediationOption> remediationOptionsOptional = Arrays.stream(remediationOptions)
```

Comment:
Should we create an internal helper for this? You're already doing it a few times, and we're going to do it a lot more as we add more of these.

Reply:
will revise.

Reply:
Updated.